In [2]:
import pandas as pd
import numpy as np

# PRECIO POR  M² URBANIZABLE

Uno de los factores fundamentales a considerar al elegir la ubicación de la empresa es el precio por metro cuadrado de terreno urbanizable en España.

Este estudio se centrará inicialmente en analizar el precio medio del metro cuadrado en las principales ciudades de las Comunidades Autónomas. Posteriormente, examinaremos el precio medio por metro cuadrado en municipios con poblaciones de entre 10,000 y 50,000 habitantes en estas mismas comunidades. 

El objetivo es identificar las variaciones en los precios entre las zonas urbanas y las áreas periféricas, lo cual proporcionará una visión clara de dónde hay mayores diferencias en costos de vivienda. Esta información será crucial, ya que nos ayudará a identificar las comunidades donde se encuentran los potenciales clientes de nuestro cliente.

Los datos se han obtenido del Ministerio de Transportes y Movilidad Sostenible, centrándose en estadísticas de precios de suelo urbano.

Para las Comunidades Autónomas utilizaremos el dataframe de precio medio en euros del metro cuadrado de suelo urbano por Comunidades Autónomas y provincias.

Para poder comenzar con el análsis hemos tenido que pasar el archivo a csv y limpiar los datos, ya que habían filas vacias y datos que no nos hacían falta, como los de los 20 años previos. Este csv se ha limpiado desde el propio excel. 

## 1. COMUNIDADES AUTÓNOMAS

### 1.1 Importamos el DF

In [3]:
#Importamos el df y le especificamos que queremos las Comunidades Autónomas como columnas para posteriormente filtrar los datos por filas.
archivo = ("df_precio_metro_cuadrado - Hoja 1.csv")
sep = (",")
header = 0

df_1 = pd.read_csv(archivo, sep= sep, header=header)
df_1.set_index("Comunidad Autonoma", inplace=True)
df_1

Primer trimestre Segundo trimestre Variacion trimestral  \
Comunidad Autonoma                                                           
total nacional                167,4             166,6                 -0,5   
Andalucia                     188,1             164,8                -12,4   
Almeria                       149,4             157,4                  5,3   
Cadiz                         193,8               217                   12   
Cordoba                         111               145                 30,7   
...                             ...               ...                  ...   
Alava                         205,1             158,6                -22,7   
Guipuzcoa                     248,2             143,3                -42,2   
Vizcaya                       302,2             221,2                -26,8   
La Rioja                        191             180,5                 -5,5   
Ceuta y Melilla                n.s.              n.s.                    .   

                   Variacion interanual  
Comunidad Autonoma                       
total nacional                      7,9  
Andalucia                          -3,3  
Almeria                           -16,3  
Cadiz                               6,9  
Cordoba                             6,2  
...                                 ...  
Alava                               -14  
Guipuzcoa                         -48,8  
Vizcaya                           -18,8  
La Rioja                           29,4  
Ceuta y Melilla                       .  

[62 rows x 4 columns]

### 1.2 Limpiamos los datos del DF para poder trabajar con ellos

In [4]:
#Reemplazamos las (,) por (.)
df_1 = df_1.replace(",", ".", regex=True)
df_1

Primer trimestre Segundo trimestre Variacion trimestral  \
Comunidad Autonoma                                                           
total nacional                167.4             166.6                 -0.5   
Andalucia                     188.1             164.8                -12.4   
Almeria                       149.4             157.4                  5.3   
Cadiz                         193.8               217                   12   
Cordoba                         111               145                 30.7   
...                             ...               ...                  ...   
Alava                         205.1             158.6                -22.7   
Guipuzcoa                     248.2             143.3                -42.2   
Vizcaya                       302.2             221.2                -26.8   
La Rioja                        191             180.5                 -5.5   
Ceuta y Melilla                n.s.              n.s.                    .   

                   Variacion interanual  
Comunidad Autonoma                       
total nacional                      7.9  
Andalucia                          -3.3  
Almeria                           -16.3  
Cadiz                               6.9  
Cordoba                             6.2  
...                                 ...  
Alava                               -14  
Guipuzcoa                         -48.8  
Vizcaya                           -18.8  
La Rioja                           29.4  
Ceuta y Melilla                       .  

[62 rows x 4 columns]

In [5]:
#Convertimos los valores de las columnas a float
df_1["Primer trimestre"] = pd.to_numeric(df_1["Primer trimestre"], errors="coerce")
df_1["Segundo trimestre"] = pd.to_numeric(df_1["Segundo trimestre"], errors="coerce")
df_1["Variacion trimestral"] = pd.to_numeric(df_1["Variacion trimestral"], errors="coerce")
df_1["Variacion interanual"] = pd.to_numeric(df_1["Variacion interanual"], errors="coerce")

df_1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 62 entries, total nacional to Ceuta y Melilla
Data columns (total 4 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Primer trimestre      61 non-null     float64
 1   Segundo trimestre     60 non-null     float64
 2   Variacion trimestral  60 non-null     float64
 3   Variacion interanual  60 non-null     float64
dtypes: float64(4)
memory usage: 2.4+ KB


### 1.3 Creamos un DF con sólo los datos con los que vamos a trabajar

In [6]:
#Seleccionamos las columnas con las que vamos a trabajar: Segundo trimestre de todas las Comunidades Autónomas y las ordenamos de menor a mayor:
df_com_auto= df_1.loc[["Andalucia", "Aragon","Asturias", "Baleares","Canarias", "Cantabria", "Castilla y Leon", "Castilla La Mancha", "Cataluña", "Comunitat Valenciana", "Extremadura", "Galicia", "Comunidad de Madrid", "Region de Murcia", "Comunidad Foral de Navarra", "Pais Vasco", "La Rioja"], ["Segundo trimestre"]]


precio_comunidades_autonomas=df_com_auto.sort_values(by= "Segundo trimestre", ascending=True)
precio_comunidades_autonomas

Segundo trimestre
Comunidad Autonoma                           
Castilla y Leon                          81.6
Castilla La Mancha                      100.4
Extremadura                             109.3
Galicia                                 112.2
Cantabria                               122.5
Aragon                                  134.2
Asturias                                135.1
Comunidad Foral de Navarra              136.0
Region de Murcia                        144.2
Andalucia                               164.8
Cataluña                                174.7
La Rioja                                180.5
Pais Vasco                              192.6
Comunitat Valenciana                    202.3
Canarias                                243.6
Comunidad de Madrid                     269.3
Baleares                                340.5

In [7]:
#Volvemos a pasar el índice que eran las Comunidades Autónomas a columna para poder visualizar los datos y trabajamos sobre este df
df1 = precio_comunidades_autonomas.reset_index()
df1

Comunidad Autonoma  Segundo trimestre
0              Castilla y Leon               81.6
1           Castilla La Mancha              100.4
2                  Extremadura              109.3
3                      Galicia              112.2
4                    Cantabria              122.5
5                       Aragon              134.2
6                     Asturias              135.1
7   Comunidad Foral de Navarra              136.0
8             Region de Murcia              144.2
9                    Andalucia              164.8
10                    Cataluña              174.7
11                    La Rioja              180.5
12                  Pais Vasco              192.6
13        Comunitat Valenciana              202.3
14                    Canarias              243.6
15         Comunidad de Madrid              269.3
16                    Baleares              340.5

### 1.4 Creación de gráficas

Para la visualización de los datos importaremos las librerías necesarias, trabajaremos con Plotly.

In [8]:
import ipywidgets as widgets
import matplotlib.pyplot as plt
import plotly.graph_objects as go

In [9]:
#Visualizamos los datos

fig = go.Figure()

fig.add_trace(go.Bar(x=df1["Comunidad Autonoma"], y=df1["Segundo trimestre"], marker_color="red"))

# Configura el layout
fig.update_layout(title="Precio medio por m² de las principales ciudades de las Comunidades Autónomas",
                  xaxis_title="Comunidades Autónomas",
                  yaxis_title="Precio medio por m²",
                  barmode='group')

fig.show()

- Gráfico 1. Precio medio por m² de las principales ciudades de las Comunidades Autónomas

### 1.5 Conclusión Comunidades Autónomas

Partimos de la siguiente premisa: las Comunidades Autónomas con los precios más bajos del m² son Castilla y León (81,6 €/m²), Castilla-La Mancha (100,4 €/m²), Extremadura (109,3 €/m²) y Galicia (112,2 €/m²). Por otro lado, las regiones con los precios más altos son Baleares (340,5 €/m²), la Comunidad de Madrid (269,3 €/m²), Canarias (243,6 €/m²) y la Comunidad Valenciana (202,3 €/m²).

Este estudio se basa en el precio medio del m² en las principales ciudades de cada Comunidad Autónoma, pero, ¿qué sucede si nos centramos en el precio del m² en las periferias de esas ciudades?

Para profundizar en este aspecto, analizaremos el precio medio del m² en suelo urbano de municipios con poblaciones de entre 10.000 y 50.000 habitantes.


## 2. COMUNIDADES AUTONOMAS. MUNICIPIOS DE 10000 Y 50000 HABITANTES

### 2.1 Importamos el DF

In [10]:
#Seguimos los mismos pasos que en el anterior:
archivo = ("df_precio_metro_cuadrado_hab - Hoja 1.csv")
sep = (",")
header = 0

df_hab1 = pd.read_csv(archivo, sep=sep, header=header)
df_hab1

df_hab1.set_index("Comunidad Autonoma", inplace=True)
df_hab1

Primer trimestre Segundo trimestre Variacion trimestral  \
Comunidad Autonoma                                                           
TOTAL NACIONAL                141,1             122,6                -13,1   
Andalucia                     123,8                90                -27,3   
Almeria                        82,3             168,6                104,9   
Cadiz                          88,2             137,5                 55,8   
Cordoba                       115,4             111,2                 -3,6   
...                             ...               ...                  ...   
Alava                             .                 .                    .   
Guipuzcoa                         .              n.s.                    .   
Vizcaya                         n.s              n.s.                    .   
La Rioja                        285             208,9                -26,7   
Ceuta y Melilla                   .                 .                    .   

                   Variacion interanual  
Comunidad Autonoma                       
TOTAL NACIONAL                      5,4  
Andalucia                         -23,6  
Almeria                            22,8  
Cadiz                              17,7  
Cordoba                            24,6  
...                                 ...  
Alava                                 .  
Guipuzcoa                             .  
Vizcaya                               .  
La Rioja                              .  
Ceuta y Melilla                       .  

[62 rows x 4 columns]

### 2.2 Limpiamos los datos del DF para poder trabajar con ellos

In [11]:
df_hab1 = df_hab1.replace(",", ".", regex=True)
df_hab1

Primer trimestre Segundo trimestre Variacion trimestral  \
Comunidad Autonoma                                                           
TOTAL NACIONAL                141.1             122.6                -13.1   
Andalucia                     123.8                90                -27.3   
Almeria                        82.3             168.6                104.9   
Cadiz                          88.2             137.5                 55.8   
Cordoba                       115.4             111.2                 -3.6   
...                             ...               ...                  ...   
Alava                             .                 .                    .   
Guipuzcoa                         .              n.s.                    .   
Vizcaya                         n.s              n.s.                    .   
La Rioja                        285             208.9                -26.7   
Ceuta y Melilla                   .                 .                    .   

                   Variacion interanual  
Comunidad Autonoma                       
TOTAL NACIONAL                      5.4  
Andalucia                         -23.6  
Almeria                            22.8  
Cadiz                              17.7  
Cordoba                            24.6  
...                                 ...  
Alava                                 .  
Guipuzcoa                             .  
Vizcaya                               .  
La Rioja                              .  
Ceuta y Melilla                       .  

[62 rows x 4 columns]

In [12]:
#Como sólo vamos a necesitar la columna "seguno trimestre" pasámos únicamente esta a float
df_hab1["Segundo trimestre"] = pd.to_numeric(df_hab1["Segundo trimestre"], errors="coerce")
df_hab1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 62 entries, TOTAL NACIONAL to Ceuta y Melilla
Data columns (total 4 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Primer trimestre      62 non-null     object 
 1   Segundo trimestre     48 non-null     float64
 2   Variacion trimestral  62 non-null     object 
 3   Variacion interanual  62 non-null     object 
dtypes: float64(1), object(3)
memory usage: 2.4+ KB


### 2.3 Creamos un DF con sólo los datos con los que vamos a trabajar

In [13]:
df_hab1 = df_hab1.loc[["Andalucia", "Aragon","Asturias", "Baleares","Canarias", "Cantabria", "Castilla y Leon", "Castilla La Mancha", "Cataluña", "Comunitat Valenciana", "Extremadura", "Galicia", "Comunidad de Madrid", "Region de Murcia", "Comunidad Foral de Navarra", "Pais Vasco", "La Rioja"], ["Segundo trimestre"]]

comunidad_auto_hab = df_hab1.sort_values(by="Segundo trimestre")
comunidad_auto_hab

Segundo trimestre
Comunidad Autonoma                           
Cantabria                                31.2
Galicia                                  65.3
Comunidad Foral de Navarra               72.8
Andalucia                                90.0
Castilla La Mancha                       91.2
Asturias                                100.0
Aragon                                  102.1
Castilla y Leon                         108.3
Extremadura                             111.0
Region de Murcia                        114.7
Pais Vasco                              125.0
Cataluña                                144.1
Comunidad de Madrid                     150.7
Comunitat Valenciana                    158.7
Canarias                                205.5
La Rioja                                208.9
Baleares                                240.1

In [14]:
df2 = comunidad_auto_hab.reset_index()
df2

Comunidad Autonoma  Segundo trimestre
0                    Cantabria               31.2
1                      Galicia               65.3
2   Comunidad Foral de Navarra               72.8
3                    Andalucia               90.0
4           Castilla La Mancha               91.2
5                     Asturias              100.0
6                       Aragon              102.1
7              Castilla y Leon              108.3
8                  Extremadura              111.0
9             Region de Murcia              114.7
10                  Pais Vasco              125.0
11                    Cataluña              144.1
12         Comunidad de Madrid              150.7
13        Comunitat Valenciana              158.7
14                    Canarias              205.5
15                    La Rioja              208.9
16                    Baleares              240.1

### 2.4 Creación de gráficas

In [15]:
#Visualizamos los datos

fig = go.Figure()

fig.add_trace(go.Bar(x=df2["Comunidad Autonoma"], y=df2["Segundo trimestre"], marker_color="orange"))

# Configura el layout
fig.update_layout(title="Precio medio por m² de los municipios de las Comunidades Autónomas entre 10000 y 50000 habitantes",
                  xaxis_title="Comunidades Autónomas",
                  yaxis_title="Precio medio por m²",
                  barmode="group")

fig.show()

- Gráfico 2. Precio medio por m² de los municipios de las Comunidades Autónomas entre 10000 y 50000 habitantes

### 2.5 Conclusión por Comunidades Autónomas de municipios de 10000 y 50000 habitantes

Del siguiente dataframe podemos observar que el precio medio por m² de las poblaciones con habitantes de entre 5000 y 10000 varia mucho en cuanto a las Comunidades Autónomas. Las zonas más económicas son Cantabria (31,2 euros/m²), Galicia (65.3 euros/m²), Comunidad de Navarra (72.8 euros/m²) y Andalucia (90 euros/m²). Mientras que las más caras siguen siendo Baleares (240 euros/m²), se suma La Rioja (208 euros/m²), Canarias (205.5 euros/m²) y la Comunidad Valenciana (158.7 euros/m²). 

### 3. COMPARACIÓN DE LOS DATOS OBTENIDOS

Ya sabemos cuánto es el precio por m² de las ciudades y de los municipios, ahora analizaremos la diferencia que hay entre ambas y para ello, nos centraemos en las Comunidades Autónomas en las que la diferencia sea siginificativa. Esto nos indicará en qué Comunidades Autónomas caro vivir en las ciudades y que por ello, las personas prefieren vivir en la periféria de las grandes ciudades. 

### 3.1 Creación de gráficos

In [17]:
fig = go.Figure()

fig.add_trace(go.Bar(x=df1["Comunidad Autonoma"], y=df1["Segundo trimestre"], name="Precio medio por m² de las Comunidades Autónomas"))
fig.add_trace(go.Bar(x=df2["Comunidad Autonoma"], y=df2["Segundo trimestre"], name= "Precio medio del m² en municipios de 10000 y 50000 habitantes "))

# Configura el layout
fig.update_layout(title="Comparación entre el precio medio por m² de las Comunidades Autónomas vs municipios de 10000 y 50000 habitantes",
                  xaxis_title="Comunidades Autónomas",
                  yaxis_title="Precio medio por m² en euros",
                  barmode='group')

fig.show()

- Gráfico 3. Comparación entre el precio medio por m² de las Comunidades Autónomas vs municipios de 10000 y 50000 habitantes

### 3.2 Conclusión

Al analizar ambos dataframes, se puede observar que el precio por metro cuadrado en las ciudades es significativamente más alto que en las áreas periféricas.

Prestando especial atención a Cantabria, notamos que en las principales ciudades el precio alcanza los 122,3 euros por metro cuadrado, mientras que en las zonas periféricas se reduce a solo 31,2 euros por metro cuadrado. Un patrón similar se encuentra en Galicia, donde el precio en las ciudades es de 112,2 euros por metro cuadrado, frente a 65,3 euros en las áreas periféricas. En Navarra, la diferencia también es notable, con un costo de 136 euros por metro cuadrado en las ciudades y 72,8 euros en la periferia.

Asimismo, en Andalucía, el precio en las ciudades es de 164,8 euros por metro cuadrado, en contraste con los 90 euros por metro cuadrado en los pueblos.

Dicha conclusión ha sido fundamental para acotar nuestra búsqueda, por lo que, para el siguiente análisis (2_climatología) nos centraremos en las dos Comunidades Autónomas anteriormente nombradas: Cantabria, Galicia, Navarra y Andalucía. 